src/server/game/Quests/QuestDef.cpp

```diff
@@ -184,48 +184,43 @@ void Quest::LoadQuestTemplateAddon(Field* fields)
         Flags |= QUEST_FLAGS_AUTO_ACCEPT;
 }
 
-uint32 Quest::XPValue(Player* player) const
+uint32 Quest::XPValue(uint8 playerLevel) const
 {
-    if (player)
-    {
-        int32 quest_level = (Level == -1 ? player->getLevel() : Level);
-        const QuestXPEntry* xpentry = sQuestXPStore.LookupEntry(quest_level);
-        if (!xpentry)
-            return 0;
-
-        int32 diffFactor = 2 * (quest_level - player->getLevel()) + 20;
-        if (diffFactor < 1)
-            diffFactor = 1;
-        else if (diffFactor > 10)
-            diffFactor = 10;
-
-        uint32 xp = diffFactor * xpentry->Exp[RewardXPDifficulty] / 10;
-        if (xp <= 100)
-            xp = 5 * ((xp + 2) / 5);
-        else if (xp <= 500)
-            xp = 10 * ((xp + 5) / 10);
-        else if (xp <= 1000)
-            xp = 25 * ((xp + 12) / 25);
-        else
-            xp = 50 * ((xp + 25) / 50);
+    int32 quest_level = (Level == -1 ? playerLevel : Level);
+    const QuestXPEntry* xpentry = sQuestXPStore.LookupEntry(quest_level);
+    if (!xpentry)
+        return 0;
 
-        return xp;
-    }
+    int32 diffFactor = 2 * (quest_level - playerLevel) + 20;
+    if (diffFactor < 1)
+        diffFactor = 1;
+    else if (diffFactor > 10)
+        diffFactor = 10;
+
+    uint32 xp = diffFactor * xpentry->Exp[RewardXPDifficulty] / 10;
+    if (xp <= 100)
+        xp = 5 * ((xp + 2) / 5);
+    else if (xp <= 500)
+        xp = 10 * ((xp + 5) / 10);
+    else if (xp <= 1000)
+        xp = 25 * ((xp + 12) / 25);
+    else
+        xp = 50 * ((xp + 25) / 50);
```

Comment:
brackets here

Reply:
I had the misfortune of editing 10-years-old code :(